orderer/consensus/smartbft/util.go

```diff
@@ -283,6 +284,22 @@ func (ri *RequestInspector) RequestID(rawReq []byte) types.RequestInfo {
 	if err != nil {
 		return types.RequestInfo{}
 	}
+
+	if req.chHdr.Type == int32(cb.HeaderType_CONFIG) {
+		configEnvelope := &cb.ConfigEnvelope{}
+		_, err = protoutil.UnmarshalEnvelopeOfType(req.envelope, cb.HeaderType_CONFIG, configEnvelope)
+		if err != nil {
+			ri.Logger.Errorf("can't get config envelope: %s", err.Error())
+			return types.RequestInfo{}
+		}
+
+		req, err = ri.unwrapReqFromEnvelop(configEnvelope.LastUpdate)
+		if err != nil {
+			ri.Logger.Errorf("can't unwrap request from last update: %s", err.Error())
+			return types.RequestInfo{}
```

Comment:
I think in the case of a config transaction, we need to hash the last update and not continue the flow, because in the normal case, we take the transaction identifier to be the request identifier, while what we really want to do is use the last update's raw bytes to be our request identifier.

Reply:
That would be good. But then we have to guarantee that there will be no map with two or more keys in the last update.

Reply:
I don't think so, because `LastUpdate` is a `common.Envelope` and is never re-marshalled inside the orderer.
The problematic stuff is the `Config` snapshot that is derived from the current config + the `LastUpdate`.

Reply:
You're right. If you want, let's do it this way.
Although I think it's unnecessary.

Reply:
@yacovm change it or leave it?

Reply:
change it please

Reply:
fix